map.js

```diff
@@ -149,69 +215,177 @@ function resizeMap(){
+            if(hoverOverlayButton == 3) currentOverlay = "Exploration";
+            drawMap();
+        }
+        else mousedown = true;
+    };
+    wrapper.onmouseup = function(){
+        mousedown = false;
+        resizeMap()
+    };
+    wrapper.onmouseout = function(){mousedown = false;};
+    wrapper.onmousemove = function(e){
+        if(mousedown){moveMap(e);}
+        
+        //Overlay mouseover
+        if(e.offsetY >= 10  && e.offsetY <= 20){
+            var x = wrapper.width;
+            if(e.offsetX >= 8 && e.offsetX <= x/3 - 1){
+                hoverOverlayButton = 1;
+                drawOverlay();
+            }
+            if(e.offsetX >= x/3 && e.offsetX <= x/3*2 - 1){
+                hoverOverlayButton = 2;
+                drawOverlay();
+            }
+            if(e.offsetX >= x/3*2 && e.offsetX <= x - 8){
+                hoverOverlayButton = 3;
+                drawOverlay();
+            }
+        } else{
+            var redraw = false;//should prevent useless redraws of the overlay when just scrolling.
+            if(hoverOverlayButton != 0){
+                redraw = true;
+            }
+
+            hoverOverlayButton = 0;
+
+            if(redraw){
+                drawOverlay();
+            }
+            //End Overlay mouseover
+
+            //mouseover icon
+            if(locArr && !mousedown && currentOverlay == "Locations"){
+                for(let i = 0; i < locArr.length;i++){
+                    
+                    let cCords = worldSpaceToCanvasSpace(locArr[i].x, locArr[i].y);
+
+                    if(cCords.x < e.offsetX &&
+                        cCords.x + cCords.wh > e.offsetX &&
+                        cCords.y < e.offsetY &&
+                        cCords.y + cCords.wh > e.offsetY){
+                            hoverLocation = locArr[i].formid;
+                            drawMap();
+                            break;
+                    }
+                    if(i == locArr.length - 1){
+                        hoverLocation= "";
+                        drawMap();
+                    }
+                }
+            }
+            //End mouseover icon
+        }
+    };
+    wrapper.onwheel = function(e){    
+        e.preventDefault();
+        if(e.deltaY > 0) zoomLevel += 0.2;
+        else zoomLevel += -0.2;
+        
+        //TODO: make it so that it zooms into middle of screen rather than top left corner?
+
+        //clamp zoom
+        if(zoomLevel > maxZoom) zoomLevel = maxZoom;
+        if(zoomLevel < minZoom) zoomLevel = minZoom;
+        moveMap();
+    };
+}
+
+//converts worldspace cords into relative canvas cords.
+//returns object{x,y} for canvas space.
+function worldSpaceToCanvasSpace(x = 0, y = 0){
+    var MapW = img_Map.width;
+    var MapH = img_Map.height;
+    var worldW = 480000;
+    var worldH = 400000;
+
+    x = (Math.round(x) + worldW / 2) / worldW;
+    y = (-Math.round(y) + worldH / 2) / worldH;
+
+    var wh = 20 / zoomLevel;
```

Comment:
the fuck is WH

Reply:
oh icon width/height. Maybe change to iconHeight. That way when we use it for text n stuff, it's like "yeah, this is the height of the icons and scaled elemenets". 
And it just so happens that the icons are square, but I don't think that we really need to specify that this is the width AND the height, we can just imply it by the fact that icons are square.

Reply:
done in next commit